File: AndroidList/app/src/main/java/com/prt2121/androidlist/MainActivity.java
package com.prt2121.androidlist;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.ListView;
import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

  private ListView listView;

  @Override protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
    listView = (ListView) findViewById(R.id.listView);
    final ItemAdapter adapter = new ItemAdapter(this);
    adapter.update(initItems());
    listView.setAdapter(adapter);
  }

  private List<Item> initItems() {
    Item item1 = new Item("Title 1", "Desc 1");
    Item item2 = new Item("Title 2", "Desc 2");
    Item item3 = new Item("Title 3", "Desc 3");
    Item item4 = new Item("Title 4", "Desc 4");
    Item item5 = new Item("Title 5", "Desc 5");
    List<Item> items = new ArrayList<>(5);
    items.add(item1);
    items.add(item2);
    items.add(item3);
    items.add(item4);
    items.add(item5);
    return items;
  }
}

File: ContentProvider/app/src/main/kotlin/com/prt2121/contentprovider/MainActivity.kt
package com.prt2121.contentprovider

import android.content.ContentValues
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log

/**
 * Created by pt2121 on 1/6/16.
 */
class MainActivity : AppCompatActivity() {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)

    val values = ContentValues()
    values.put(InviteEntry.INVITE_ID, "12349")
    values.put(InviteEntry.TO_USER, "prat")
    values.put(InviteEntry.TO_USER_PROFILE_IMAGE_URI, "uri")
    values.put(InviteEntry.DESTINATION_LATLNG, "pickup latlng")
    values.put(InviteEntry.DESTINATION_ADDRESS, "pickup address")
    values.put(InviteEntry.MESSAGE, "hello there")
    values.put(InviteEntry.STATUS, "PENDING")
    values.put(InviteEntry.PICKUP_ADDRESS, "pickup address")

    val uri = contentResolver.insert(InviteContract.CONTENT_URI, values)
    Log.v("MainActivity", "uri ${uri.toString()}")
    if(uri != null) {
      val cursor = contentResolver.query(InviteContract.CONTENT_URI, null, null, null, null)
      //val cursor = contentResolver.query(uri, null, null, null, null)
      if(cursor != null) {
        cursor.moveToFirst()
        val status = cursor.getString(cursor.getColumnIndexOrThrow(InviteEntry.STATUS))
        val id = cursor.getInt(cursor.getColumnIndexOrThrow(InviteEntry._ID))
        Log.v("MainActivity", "id $id status $status")
      }
    }
  }
}